openwisp_ipam/base/models.py

```diff
@@ -212,13 +225,10 @@ def _read_ipaddress_data(self, reader, subnet):
             ip.save()
 
     def _get_csv_reader(self, file):
-        if file.name.endswith(('.xls', '.xlsx')):
-            book = xlrd.open_workbook(file_contents=file.read())
-            sheet = book.sheet_by_index(0)
-            row = []
-            for row_num in range(sheet.nrows):
-                row.append(sheet.row_values(row_num))
-            reader = iter(row)
+        if file.name.endswith(('.xlsx')):
+            book = openpyxl.load_workbook(filename=file)
+            sheet = book.worksheets[0]
+            reader = sheet.values
```

Comment:
Earlier the project was using [xlrd](https://xlrd.readthedocs.io/en/latest/changes.html#id1) package for reading xls and xlsx files. But the latest version of xlrd now only processes xls files. Hence, I swapped the package for [openxlpy](https://openpyxl.readthedocs.io/) which can read xlsx files (but not xls files). I changed the dependency with the thought that more and more people will be using xlsx files in the current time. 

We can look for other alternatives if we really want to support xls files.

Reply:
if we can do csv that's fine

Reply:
Right now we can do CSV and XLSX. 

The library for parsing CSV is ships with Python. (no external library/package is required)
For parsing XLSX we will be using [openxlpy](https://openpyxl.readthedocs.io/)

With this PR, we will be dropping support for XLS. Should we also drop support for XLSX?

Reply:
it's fine as it is, let's not touch this further right now, there's no reason to.